kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ResourceSelection.vue

```diff
@@ -271,7 +273,17 @@
       isSelectAllChecked() {
         // Returns true if all the resources in the topic are in the working resource pool
         const workingResourceIds = this.workingResourcePool.map(wr => wr.id);
-        return this.contentList.every(content => workingResourceIds.includes(content.id));
+
+        if (this.contentList.every(content => workingResourceIds.includes(content.id))) {
+          this.contentList.forEach(resource => {
+            if (resource.kind === ContentNodeKinds.TOPIC) {
```

Comment:
I think that maybe this logic is altogether not necessary here. It seems like you're aiming to handle the user's "select all" click while also including all of the folders' children in that "Select all".

The logic for "selecting folders" can be handled in a separate issue, so this should be reverted to the `return this.contentList.every(...)`.

One thing to bear in mind is that even when we do allow users to select folders, the only time we'll place all of the topic's children into the working resource list is if that folder meets the criteria for having a checkbox in the first place as defined in #11790